fs-storage/src/file_storage.rs

```diff
@@ -224,13 +234,54 @@ where
     }
 }
 
+/// Combine two `FileStorage` instances into a new instance.
+pub fn combine_file_storages<K, V>(
+    a: &FileStorage<K, V>,
+    b: &FileStorage<K, V>,
+) -> FileStorage<K, V>
```

Comment:
Let's move this function to `BaseStorage` trait:
```
    fn merge_from(&mut self, other: AsRef<BTreeMap<K, V>>) -> Result<()>;
```

This makes it explicit what path is used to store the result. Also, making it a function in the trait allows very flexible usage. For isntance, we could update a file storage using values from a folder storage (or future chunked storage).